acme/acme/client_test.py

```diff
@@ -789,6 +796,18 @@ def test_revoke(self):
         self.net.post.assert_called_once_with(
             self.directory["revokeCert"], mock.ANY, acme_version=2)
 
+    def test_update_registration(self):
+        # "Instance of 'Field' has no to_json/update member" bug:
+        # pylint: disable=no-member
+        self.response.headers['Location'] = self.regr.uri
```

Comment:
:+1:

Reply:
that change won't actually do that, but I'll do the thing that wil

Reply:
actually, this is really hard to do on this testing infrastructure, since we usually just mock out network requests, and filling in a different location header also updates it for update_registration

so what we actually want to do is check that the second time we call out to the network, we call with the url we stuck in there. that's also not set up in this infrastructure to check (mock doesn't seem to have a "first time you called with this, second time you called with that" functionality).

so I'm going to change this to just check that we called `_get_v2_account`. lmk if you have a different way you'd like to do it; otherwise, doesn't seem like a unit test is the best test here.

Reply:
*check that post was called twice, which implies that we called `_get_v2_account`

Reply:
> filling in a different location header also updates it for update_registration

If we wanted to change this I think we could do something like `self.net.post.side_effect = [response1, response2]`, but I agree with you that this probably isn't the best way to test this.

> so what we actually want to do is check that the second time we call out to the network, we call with the url we stuck in there. that's also not set up in this infrastructure to check (mock doesn't seem to have a "first time you called with this, second time you called with that" functionality).

I think you could use:
```
uri_used = self.net.post.call_args[0][0]
```
This returns the first positional argument given in the most recent call to `self.net.post`.

Alternatively you could use:
```
uri_used = self.net.post.call_args_list[1][0][0]
```
which would be the first positional argument given in the second call of `self.net.post`.

We could also use an approach like this to test that `only_return_existing` is set to `True` and/or the URI used in the first call of `post`.

I'm not going to block landing this PR on making these changes, but I think it might be worthwhile.

Reply:
I think it's a pretty marginal improvement over just checking that post was called twice, but I'll add `self.assertTrue(DIRECTORY_V2.newAccount in self.net.post.call_args_list[0][0])`

Reply:
Thanks!